src/ophys_etl/transforms/sine_dewarp.py

```diff
@@ -0,0 +1,459 @@
+
+    aLstr = data.get("aL", None)
+    if aLstr is None:
+        raise KeyError(f"parameter aL not determined for {equipment_name}")
+    aL = float(aLstr)
+
+    aRstr = data.get("aR", None)
+    if aRstr is None:
+        raise KeyError(f"parameter aR not determined for {equipment_name}")
+    aR = float(aRstr)
+
+    bLstr = data.get("bL", None)
+    if bLstr is None:
+        raise KeyError(f"parameter bL not determined for {equipment_name}")
+    bL = float(bLstr)
+
+    bRstr = data.get("bR", None)
+    if bRstr is None:
+        raise KeyError(f"parameter bR not determined for {equipment_name}")
+    bR = float(bRstr)
+
+    print(str(equipment_name))
+    print('aL bL', aL, bL)
+    print('aR bR', aR, bR)
+
+    return input_h5, output_h5, aL, aR, bL, bR
+
+
+def run_dewarping(FOVwidth, noise_reduction, threads,
+                  input_file, input_dataset,
+                  output_file, output_dataset):
+    """
+
+    """
+
+    # Get statistics about the movie
+    input_h5_file = h5py.File(input_file, 'r')
+    movie = input_h5_file[input_dataset]
+    meanimg = np.mean(movie, axis=0)  # use avgimg to compute mode
+    meanimg = meanimg/256          # less noisy to get mode from 8bit histogram
+    meanimg = meanimg.astype(int)  # less noisy to get mode from 8bit histogram
+
+    # full movie got Memory Error
+    stdvimg = np.std(movie[::8], axis=0)  # use avgimg to compute mode
+    stdvimg = stdvimg/256          # less noisy to get mode from 8bit histogram
+    stdvimg = stdvimg.astype(int)  # less noisy to get mode from 8bit histogram
+
+    movie_shape = movie.shape
+    movie_dtype = movie.dtype
+    T, y_orig, x_orig = movie.shape
+
+    # need one frame from movie for mode
+    # IMPORTANT!: estmated modevalue is from 8bit img but convert back to 16bit
+    xtable = create_xtable(meanimg, stdvimg, aL, aR, bL, bR, noise_reduction)
+
+    # Remove old output if it exists and create new output file
+    if os.path.exists(output_file):
+        os.remove(output_file)
+
+    dewarped_file = h5py.File(output_file, 'w')
+    if FOVwidth == 512:
+        dewarped_file.create_dataset(
+            output_dataset, shape=movie_shape, dtype=movie_dtype
+        )
+    else:
+        out_shape = [movie_shape[0], movie_shape[1], 512 - R - L]
+        dewarped_file.create_dataset(
+            output_dataset, shape=out_shape, dtype=movie_dtype
+        )
+    dewarped_file.close()
+
+    start_time = time.time()
+
+    # Separate frames into elementes of a list and dewarp each indvidually
+    images = [
+        input_h5_file[input_dataset][frame, :, :] for frame in range(T)
+    ]
+
+    with multiprocessing.Pool(threads) as pool:
+        fn = functools.partial(
+            xdewarp,
+            FOVwidth=FOVwidth,
+            xtable=xtable,
+            noise_reduction=noise_reduction
+        )
+        result = pool.map_async(fn, images, chunksize=1)
+
+        while not result.ready():
```

Comment:
If I were to refactor this I'd use a callback function on line 392 for the `map_async` call to report the results as they come in, rather than polling every 5 seconds